src/pages/VenueLandingPage/VenueLandingPage.tsx

```diff
@@ -91,6 +104,56 @@ export const VenueLandingPage: React.FunctionComponent<VenueLandingPageProps> =
     }
   }, [shouldOpenPaymentModal, isAuthenticationModalOpen]);
 
+  const onJoinClick = useCallback(() => {
+    if (venueId) {
+      const venueEntrance = venue?.entrance && venue.entrance.length;
+      window.location.href =
+        user && !venueEntrance
+          ? venueInsideUrl(venueId)
+          : venueEntranceUrl(venueId);
+    }
+  }, [user, venue, venueId]);
+
+  const renderJoinButton = useCallback(() => {
+    switch (venue?.accessMode) {
+      case VenueAccessMode.password:
+        return (
+          <div className="secret-password-form-wrapper">
+            <SecretPasswordForm
+              buttonText={
+                venue.config?.landingPageConfig.joinButtonText ??
+                "Join the party"
+              }
+              onSuccess={showJoinVenueModal}
+            />
+          </div>
+        );
+
+      case VenueAccessMode.email:
+      case VenueAccessMode.ticket:
+      case VenueAccessMode.parent:
+      case VenueAccessMode.open:
+      default:
```

Comment:
Since you're switching on `venue?.accessMode`, this `default` case will also match when `venue` is `undefined`